scripts/doccy.coffee

```diff
@@ -29,44 +29,64 @@ module.exports = (robot) ->
+
   learnMethod = (res) ->
-    key = res.match[1]
-    value = res.match[2]
-    robot.brain.thoughts[key] = value
-    res.reply "gotcha, '#{key}' means '#{value}'"
+    [_, key, value] = res.match
+    if thoughts[key]
+      res.reply "I've already learnt that \" #{key} \" means #{thoughts[key]}"
+    else
+      actuallyLearnMethod(res, key, value)
+
+  relearnMethod = (res) ->
+    [_, key, value] = res.match
+    actuallyLearnMethod(res, key, value)
 
-  robot.hear /^!learn "(.*)" (.*)$/i, (res)-> learnMethod(res)
-  robot.respond /learn "(.*)" (.*)$/i, (res)-> learnMethod(res)
+  robot.hear /^!learn "([^"]+)" (.+)$/i, (res)-> learnMethod(res)
+  robot.respond /learn "([^"]+)" (.+)$/i, (res)-> learnMethod(res)
+
+  robot.respond /relearn "([^"]+)" (.+)$/i, (res)-> relearnMethod(res)
 
   robot.respond /\? learn$/, (res) ->
-    res.send "* @tombot learn \"quotes-delimited key\" the thing you want me to remember"
-    res.send "* !learn \"quotes-delimited key\" the things you want me to remember"
+    res.send "> @tombot learn \"quotes-delimited key\" the thing you want me to remember"
+    res.send "> !learn \"quotes-delimited key\" the things you want me to remember"
     res.reply "I will learn something new for you."
 
+  robot.respond /\? relearn$/, (res) ->
+    res.send "* @tombot relearn \"quotes-delimited key\" the thing you want me to remember"
+    res.reply "I will replace what I have learned with the new information."
+
   rememberMethod = (res) ->
-    match = res.match[1]
-    if `match in robot.brain.thoughts`
-      res.send robot.brain.thoughts[match]
+    [_, match] = res.match
+    if `match in thoughts`
+      res.send thoughts[match]
     else
       res.send "sorry, I don't know this :("
 
-  robot.hear /^!remember "(.*)"/, rememberMethod
-  robot.respond /.*remember "(.*)".*/, rememberMethod
+  robot.hear /^!remember "([^"]+)"/, rememberMethod
+  robot.respond /.*remember "([^"]+)".*/, rememberMethod
 
   robot.respond /\? remember/, (res) ->
-    res.reply "* @tombot remember \"quotes-delimited key\"
-      * !remember \"quotes-delimited key\"
-      I will try to remember something for you."
+    res.send "> `@tombot remember \"quotes-delimited key`\""
+    res.send "> `!remember \"quotes-delimited key`\""
+    res.respond "I will try to remember something for you."
 
   robot.respond /learned/, (res) ->
     res.send "Here's what I learned:"
-    res.send '"' + thought + '" ' + robot.brain.thoughts[thought] for thought of robot.brain.thoughts
+    res.send "*\"#{thought}\"*: #{thoughts[thought]}" for thought of thoughts
     res.emote "Fin"
 
   robot.respond /\? learned$/, (res) ->
     res.send " * learned // in a Direct Message"
     res.reply "I will tell you every single thing I've learned."
     res.reply "Since it might be a lot please only ask me this in a Direct Message"
+
+  robot.brain.on 'loaded', ->
+    thoughts = robot.brain.data.thoughts
+    thoughts ?= {}
```

Comment:
well, it's working xD I think it doesn't matter since it has to load the entire script and then wait for the events